src/utilities/geometry/Plane.hpp

```diff
@@ -102,6 +104,13 @@ class UTILITIES_API Plane
   /// coefficient d
   double d() const;
 
+  // Returns any Point3d that's on this plane
+  Point3d anyPointOnPlane() const;
+
+  /** Project a directed ray towards a plane and returns the intersection point if any. The RayDirection should be shooting TOWARDS the plane
+    * If enforceDirectionOfPlane is true, the ray direction and the plane outward normal must be facing each other */
+  boost::optional<Point3d> rayIntersection(const Point3d& rayOrigin, const Vector3d& rayDirection, bool enforceDirectionOfPlane = false) const;
```

Comment:
New Plane methods for getting any point3d on a plane, and to calculate the intersection point with a directed ray